app/ui-cached-collection/client/models/CachedCollection.js

```diff
@@ -177,157 +175,77 @@ export class CachedCollection {
 
-			if (this.updatedAt < new Date()) {
-				this.updatedAt = new Date();
+			if (_updatedAt > this.updatedAt) {
+				this.updatedAt = _updatedAt;
 			}
-
-			callback(data);
 		});
-	}
 
-	loadFromServerAndPopulate() {
-		this.loadFromServer((loadedData) => {
-			this.ready.set(true);
-			this.saveCache(loadedData);
-		});
-	}
-
-	sync() {
-		if (CachedCollectionManager.syncEnabled === false || Meteor.connection._outstandingMethodBlocks.length !== 0) {
-			return false;
-		}
-
-		this.log(`syncing from ${ this.updatedAt }`);
-
-		Meteor.call(this.syncMethodName, this.updatedAt, (error, data) => {
-			let changes = [];
-
-			if (data.update && data.update.length > 0) {
-				this.log(`${ data.update.length } records updated in sync`);
-				changes.push(...data.update);
-			}
+		this.collection._collection._docs._map = fromEntries(data.records.map((record) => [record._id, record]));
+		this.updatedAt = data.updatedAt;
 
-			if (data.remove && data.remove.length > 0) {
-				this.log(`${ data.remove.length } records removed in sync`);
-				changes.push(...data.remove);
-			}
-
-			changes = changes.sort((a, b) => {
-				const valueA = a._updatedAt || a._deletedAt;
-				const valueB = b._updatedAt || b._deletedAt;
-
-				if (valueA < valueB) {
-					return -1;
-				}
-
-				if (valueA > valueB) {
-					return 1;
-				}
-
-				return 0;
-			});
-
-			for (const record of changes) {
-				callbacks.run(`cachedCollection-sync-${ this.name }`, record, record._deletedAt ? 'removed' : 'changed');
-				if (record._deletedAt) {
-					this.collection.remove({ _id: record._id });
-
-					this.onSyncData('removed', record);
+		_.each(this.collection._collection.queries, (query) => this.collection._collection._recomputeResults(query));
+	}
 
-					if (record._deletedAt && record._deletedAt > this.updatedAt) {
-						this.updatedAt = record._deletedAt;
-					}
-				} else {
-					this.collection.upsert({ _id: record._id }, _.omit(record, '_id'));
+	async loadFromServer() {
+		const startTime = new Date();
+		const lastTime = this.updatedAt;
+		const data = await call(this.methodName);
+		this.log(`${ data.length } records loaded from server`);
+		data.forEach((record) => {
+			callbacks.run(`cachedCollection-loadFromServer-${ this.name }`, record, 'changed');
 
-					this.onSyncData('changed', record);
+			this.collection.direct.upsert({ _id: record._id }, _.omit(record, '_id'));
 
-					if (record._updatedAt && record._updatedAt > this.updatedAt) {
-						this.updatedAt = record._updatedAt;
-					}
-				}
-			}
+			this.onSyncData('changed', record);
 
-			this.saveCache();
+			if (record._updatedAt && record._updatedAt > this.updatedAt) { this.updatedAt = record._updatedAt; }
 		});
-
-		return true;
+		this.updatedAt = this.updatedAt === lastTime ? startTime : this.updatedAt;
 	}
 
-	saveCache(data) {
-		if (this.useCache === false) {
-			return;
-		}
+	async loadFromServerAndPopulate() {
+		await this.loadFromServer();
+		this.save();
```

Comment:
Is it safe to skip `this.save()` after `await this.loadFromServer()` throws any error?

Reply:
should be, if we are loading from server is because we dont have any record, if we failed trying to receive anything from server we dont have anything to save..